deployment/clouddeploy/deployer.go

```diff
@@ -561,32 +932,83 @@ func (d *Deployer) ModifyCluster(ctx context.Context, clusterID string, def *clu
 		return errors.Wrap(err, "failed to build cluster specs")
 	}
 
-	d.logger.Debug("generated new specification list", zap.Any("specs", newSpecs))
+	if !isServiceEqual(clusterInfo.Cluster.Services, newSpecs) {
+		d.logger.Info("cluster current spec is different from the def spec")
+		d.logger.Debug("generated new specification list", zap.Any("specs", newSpecs))
+		err = d.client.UpdateClusterSpecs(
+			ctx,
+			d.tenantID,
+			cloudProjectID,
+			cloudClusterID,
+			&capellacontrol.UpdateClusterSpecsRequest{
+				Specs: newSpecs,
+			})
+		if err != nil {
+			return errors.Wrap(err, "failed to update cluster specs")
+		}
 
-	err = d.client.UpdateClusterSpecs(
-		ctx,
-		d.tenantID,
-		cloudProjectID,
-		cloudClusterID,
-		&capellacontrol.UpdateClusterSpecsRequest{
-			Specs: newSpecs,
-		})
-	if err != nil {
-		return errors.Wrap(err, "failed to update cluster specs")
-	}
+		d.logger.Debug("waiting for cluster modification to begin")
+
+		err = d.mgr.WaitForClusterState(ctx, d.tenantID, cloudClusterID, "scaling")
+		if err != nil {
+			return errors.Wrap(err, "failed to wait for cluster modification to begin")
+		}
 
-	d.logger.Debug("waiting for cluster modification to begin")
+		d.logger.Debug("waiting for cluster to be healthy")
 
-	err = d.mgr.WaitForClusterState(ctx, d.tenantID, cloudClusterID, "scaling")
-	if err != nil {
-		return errors.Wrap(err, "failed to wait for cluster modification to begin")
+		err = d.mgr.WaitForClusterState(ctx, d.tenantID, cloudClusterID, "healthy")
+		if err != nil {
+			return errors.Wrap(err, "failed to wait for cluster to be healthy")
+		}
 	}
 
-	d.logger.Debug("waiting for cluster to be healthy")
+	// Server version can only be modified in cloud deployer
+	if def.Deployer == "cloud" {
```

Comment:
This is the `clouddeploy.Deployer`, so you don't need to check the deployer type here.

Reply:
yes, silly mistake, updated